ckanext/datastore/blueprint.py

```diff
@@ -141,11 +148,7 @@ class DictionaryView(MethodView):
 
     def _prepare(self, id: str, resource_id: str) -> dict[str, Any]:
         try:
-            check_access(
-                "datastore_create",
-                context={"user": current_user.name, "auth_user_obj": current_user},
-                data_dict={"resource_id": resource_id},
-            )
+            h.check_access("datastore_create", {"resource_id": resource_id})
```

Comment:
This is incorrect.  Calling h.check_access and not checking the return boolean essentially bypasses the check.